habitat/plan.sh

```diff
@@ -0,0 +1,208 @@
+pkg_description="Chef Workstation - Opinionated tools for getting the most out of the Chef ecosystem"
+pkg_license=('Apache-2.0')
+pkg_bin_dirs=(bin)
+pkg_svc_user=root
+ruby_pkg="core/ruby26"
+pkg_build_deps=(
+  core/make
+  core/gcc
+  core/go
+  core/gcc-libs
+  core/pkg-config
+  # We make this a build dependency since we are going to import
+  # the generated binary into our list of binaries.
+  # @afiune: Not sure if this is the right pattern but it works.
+  chef/chef-analyze
+)
+pkg_deps=(
+  core/glibc
+  core/bash
+  ${ruby_pkg}
+  core/libxml2
+  core/libxslt
+  core/xz
+  core/zlib
+  core/bundler
+  core/openssl
+  core/cacerts
+  core/libffi
+  core/libarchive
+  core/coreutils
+  core/git
+)
+
+pkg_version() {
+  cat "${SRC_PATH}/VERSION"
+}
+
+do_before() {
+  do_default_before
+  update_pkg_version
+}
+
+do_verify() {
+  return 0
+}
+
+do_prepare() {
+  export OPENSSL_LIB_DIR=$(pkg_path_for openssl)/lib
+  export OPENSSL_INCLUDE_DIR=$(pkg_path_for openssl)/include
+  export SSL_CERT_FILE=$(pkg_path_for cacerts)/ssl/cert.pem
+  export RUBY_ABI_VERSION=$(ls $(pkg_path_for ${ruby_pkg})/lib/ruby/gems)
+  build_line "Using Ruby ABI version '${RUBY_ABI_VERSION}'"
+
+  build_line "Setting link for /usr/bin/env to 'coreutils'"
+  if [ ! -f /usr/bin/env ]; then
+    ln -s "$(pkg_interpreter_for core/coreutils bin/env)" /usr/bin/env
+  fi
+}
+
+do_build() {
+  export CPPFLAGS="${CPPFLAGS} ${CFLAGS}"
+  export GEM_HOME
+  export GEM_PATH
+  # TODO this appears to give us no depsolver? What are the effects?
+  GEM_HOME="$pkg_prefix"
+  GEM_PATH="$(pkg_path_for bundler):${GEM_HOME}"
+
+  export NOKOGIRI_CONFIG
+  NOKOGIRI_CONFIG="--use-system-libraries \
+    --with-zlib-dir=$(pkg_path_for zlib) \
+    --with-xslt-dir=$(pkg_path_for libxslt) \
+    --with-xml2-include=$(pkg_path_for libxml2)/include/libxml2 \
+    --with-xml2-lib=$(pkg_path_for libxml2)/lib \
+    --without-iconv"
+
+  ( cd "${SRC_PATH}/components/gems" || exit_with "unable to enter components/gems directory" 1
+    bundle config --local build.nokogiri "$NOKOGIRI_CONFIG"
+    bundle config --local silence_root_warning 1
+    bundle install --without dep_selector --no-deployment --jobs 10 --retry 5 --path "$pkg_prefix"
+  )
+
+  build_line "Building top-level 'chef' CMD wrapper"
+  ( cd "${SRC_PATH}/components/main-chef-wrapper" || exit_with "unable to enter main-chef-wrapper directory" 1
+    CGO_ENABLED=0 go build -o "$pkg_prefix/bin/chef"
+  )
+}
+
+#######################################################
+# !!!              IMPORTANT REMINDER             !!! #
+#######################################################
+# Any changes to plan.sh related to installed gems    #
+# (eg 'without' flags, additions/removals) must       #
+# also be updated in omnibus/config/software/gems.rb  #
+#######################################################
+
+do_install() {
+  export ruby_bin_dir
+  ruby_bin_dir="$pkg_prefix/ruby-bin"
+
```

Comment:
I like this better, we're doing something similar now in the WS omnibus build too.